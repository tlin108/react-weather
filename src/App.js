import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="site-content">
          <div className="site-header">
            <div className="container">
              <a href="index.html" className="branding">
                <img src="images/logo.png" alt="" className="logo" />
                <div className="logo-type">
                  <h1 className="site-title">Company name</h1>
                  <small className="site-description">tagline goes here</small>
                </div>
              </a>
            </div>
          </div>

          <div className="hero" data-bg-image="images/banner.png">
            <div className="container">
              <form action="#" className="find-location">
                <input type="text" placeholder="Find your location..." />
                <input type="submit" value="Find" />
              </form>
            </div>
          </div>
          <div className="forecast-table">
            <div className="container">
              <div className="forecast-container">
                <div className="today forecast">
                  <div className="forecast-header">
                    <div className="day">Monday</div>
                    <div className="date">6 Oct</div>
                  </div>
                  <div className="forecast-content">
                    <div className="location">New York</div>
                    <div className="degree">
                      <div className="num">23<sup>o</sup>C</div>
                      <div className="forecast-icon">
                        <img src="images/icons/icon-1.svg" alt="" />
                      </div>  
                    </div>
                    <span><img src="images/icon-umberella.png" alt="" />20%</span>
                    <span><img src="images/icon-wind.png" alt="" />18km/h</span>
                    <span><img src="images/icon-compass.png" alt="" />East</span>
                  </div>
                </div>
                <div className="forecast">
                  <div className="forecast-header">
                    <div className="day">Tuesday</div>
                  </div>
                  <div className="forecast-content">
                    <div className="forecast-icon">
                      <img src="images/icons/icon-3.svg" alt=""/>
                    </div>
                    <div className="degree">23<sup>o</sup>C</div>
                    <small>18<sup>o</sup></small>
                  </div>
                </div>
                <div className="forecast">
                  <div className="forecast-header">
                    <div className="day">Wednesday</div>
                  </div> 
                  <div className="forecast-content">
                    <div className="forecast-icon">
                      <img src="images/icons/icon-5.svg" alt="" />
                    </div>
                    <div className="degree">23<sup>o</sup>C</div>
                    <small>18<sup>o</sup></small>
                  </div>
                </div>
                <div className="forecast">
                  <div className="forecast-header">
                    <div className="day">Thursday</div>
                  </div> 
                  <div className="forecast-content">
                    <div className="forecast-icon">
                      <img src="images/icons/icon-7.svg" alt=""/>
                    </div>
                    <div className="degree">23<sup>o</sup>C</div>
                    <small>18<sup>o</sup></small>
                  </div>
                </div>
                <div className="forecast">
                  <div className="forecast-header">
                    <div className="day">Friday</div>
                  </div> 
                  <div className="forecast-content">
                    <div className="forecast-icon">
                      <img src="images/icons/icon-12.svg" alt=""/>
                    </div>
                    <div className="degree">23<sup>o</sup>C</div>
                    <small>18<sup>o</sup></small>
                  </div>
                </div>
                <div className="forecast">
                  <div className="forecast-header">
                    <div className="day">Saturday</div>
                  </div> 
                  <div className="forecast-content">
                    <div className="forecast-icon">
                      <img src="images/icons/icon-13.svg" alt=""/>
                    </div>
                    <div className="degree">23<sup>o</sup>C</div>
                    <small>18<sup>o</sup></small>
                  </div>
                </div>
                <div className="forecast">
                  <div className="forecast-header">
                    <div className="day">Sunday</div>
                  </div> 
                  <div className="forecast-content">
                    <div className="forecast-icon">
                      <img src="images/icons/icon-14.svg" alt=""/>
                    </div>
                    <div className="degree">23<sup>o</sup>C</div>
                    <small>18<sup>o</sup></small>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
