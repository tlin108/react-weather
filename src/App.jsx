import React, { Component } from 'react';
import './App.css';

import Header from './components/Header';
import SearchBar from './components/SearchBar';
import ForcastTable from './components/ForcastTable';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="site-content">
          <Header />

          <div className="hero" data-bg-image="images/banner.png">
            <div className="container">
              <SearchBar />
            </div>
          </div>

          <ForcastTable />
        </div>
      </div>
    );
  }
}

export default App;
