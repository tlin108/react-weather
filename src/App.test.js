import React from 'react';
import { mount, shallow } from 'enzyme';
import App from './App';
import sinon from 'sinon';
import nock from 'nock';

const mockWeatherList = {
      list:
        [{
          dt: 1492185600,
          temp: {
            min: 49.77,
            max: 49.77
          },
          weather: [{
            icon: "01d"
          }],
          humidity: 69,
          speed: 1.97
        }, {
          dt: 1492185700,
          temp: {
            min: 49.77,
            max: 49.77
          },
          weather: [{
            icon: "01d"
          }]
        }, {
          dt: 1492185800,
          temp: {
            min: 49.77,
            max: 49.77
          },
          weather: [{
            icon: "01d"
          }]
        }, {
          dt: 1492185900,
          temp: {
            min: 49.77,
            max: 49.77
          },
          weather: [{
            icon: "01d"
          }]
        }, {
          dt: 1492186000,
          temp: {
            min: 49.77,
            max: 49.77
          },
          weather: [{
            icon: "01d"
          }]
        }]
      };

describe('App test suite',() => {
  let server;
  let wrapper;

  beforeEach(() => {
    server = sinon.fakeServer.create();
    var response = [200, {'Content-type': 'application/json'}, JSON.stringify(mockWeatherList)];
    server.respondWith('GET', 'http://api.openweathermap.org/data/2.5/forecast/daily?q=New%20York&units=imperial&cnt=5&appid=d22aa35f1d715ba35838639d67d4cff8', response);
  });

  afterEach(() => {
    server.restore();
  });

  it('renders without crashing', () => {
    wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('calls `componentDidMount` lifecycle method', () => {
    sinon.spy(App.prototype, 'componentDidMount');
    wrapper = mount(<App />);
    expect(App.prototype.componentDidMount.calledOnce).toEqual(true);
  });

  it('fetches weather data', () => {
    wrapper = mount(<App />);
    // have to explicitly setState because setState isnt triggered in fetch method
    // when mounted by enzyme, need to find a way around
    wrapper.setState({ weatherList: mockWeatherList.list});
    expect(wrapper.state().weatherList).toBeInstanceOf(Array);
    expect(wrapper.state().weatherList.length).toEqual(5);
  });

  it('calls `fetchWeather` method after state.searchCity changed', () => {
    wrapper = mount(<App />);
    expect(wrapper.state().searchCity).toEqual('New York');
    // spy on fetchWeathers method
    sinon.spy(wrapper.instance(), "fetchWeathers");

    wrapper.setState({ searchCity: 'Boston' });
    expect(wrapper.state().searchCity).toEqual('Boston');
    expect(wrapper.instance().fetchWeathers.calledOnce).toEqual(true);
  });
});

export default mockWeatherList;
