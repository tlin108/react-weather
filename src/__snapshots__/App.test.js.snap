exports[`App test suite renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App">
    <div
        className="site-content">
        <Header />
        <div
            className="hero"
            data-bg-image="images/banner.png">
            <div
                className="container">
                <SearchBar
                    onSearchSubmit={[Function]} />
            </div>
        </div>
        <ForecastTable
            city="New York"
            error={false}
            weatherList={Array []} />
    </div>
</div>,
  "nodes": Array [
    <div
      className="App">
      <div
            className="site-content">
            <Header />
            <div
                  className="hero"
                  data-bg-image="images/banner.png">
                  <div
                        className="container">
                        <SearchBar
                              onSearchSubmit={[Function]} />
                  </div>
            </div>
            <ForecastTable
                  city="New York"
                  error={false}
                  weatherList={Array []} />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "fetchWeathers": [Function],
        "onSearchCityChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "fetchError": false,
          "searchCity": "New York",
          "weatherList": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App">
          <div
                    className="site-content">
                    <Header />
                    <div
                              className="hero"
                              data-bg-image="images/banner.png">
                              <div
                                        className="container">
                                        <SearchBar
                                                  onSearchSubmit={[Function]} />
                              </div>
                    </div>
                    <ForecastTable
                              city="New York"
                              error={false}
                              weatherList={Array []} />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="App">
          <div
                    className="site-content">
                    <Header />
                    <div
                              className="hero"
                              data-bg-image="images/banner.png">
                              <div
                                        className="container">
                                        <SearchBar
                                                  onSearchSubmit={[Function]} />
                              </div>
                    </div>
                    <ForecastTable
                              city="New York"
                              error={false}
                              weatherList={Array []} />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
